python_files/generate_supplementary_plots.py

```diff
@@ -13,6 +23,82 @@
 import seaborn as sns
 
 
+def stitch_and_annotate_padded_img(image_data: xr.DataArray, padding: int = 25,
+                                   font_size: int = 100):
+    """Stitch an image with (x, y, z) dimensions, and annotate each image with labels contained
+    in the xth dimension.
+
+    Args:
+        image_data (xr.DataArray):
+            The image data to tile, should be 3D
+        padding (int):
+            Amount of padding to add around each channel in the stitched image
+        font_size (int):
+            The font size to use for annotations
+
+    Returns:
+        Image:
+            The PIL image instance which contains the stitched (with padding) and annotated image
+    """
+    # param validation
+    if padding <= 0:
+        raise ValueError("padding must be a positive integer")
+    if font_size <= 0:
+        raise ValueError("font_size must be a positive integer")
+
+    # define the number of rows and columns
+    num_cols: int = math.isqrt(image_data.shape[0])
+    num_rows: int = math.ceil(image_data.shape[0] / num_cols)
+    row_len: int = image_data.shape[1]
+    col_len: int = image_data.shape[2]
+
+    # create the blank image, start with a fully white slate
+    stitched_image: np.ndarray = np.zeros(
+        (
+            num_rows * row_len + (num_rows - 1) * padding,
+            num_cols * col_len + (num_cols - 1) * padding
+        )
+    )
+    stitched_image.fill(255)
+
+    # retrieve the annotation labels
+    annotation_labels: List[Any] = list(image_data.coords[image_data.dims[0]].values)
+
+    # stitch the channels
+    img_idx: int = 0
+    for row in range(num_rows):
+        for col in range(num_cols):
+            stitched_image[
+                (row * row_len + padding * row) : ((row + 1) * row_len + padding * row),
+                (col * col_len + padding * col) : ((col + 1) * col_len + padding * col)
+            ] = image_data[img_idx, ...]
+            img_idx += 1
+            if img_idx == len(annotation_labels):
+                break
+
+    # define a draw instance for annotating the channel name
+    stitched_image_im: Image = Image.fromarray(stitched_image)
```

Comment:
How is PIL? Do you find it easy to use?

Reply:
Haven't used it too much, it's nice for basic image manipulation and analysis such as simple morphology and filtering. IMO though it's not the easiest for more complex analysis, especially conversions. I've found `opencv` easier for even tasks such as converting a grayscale to RGB.